src/alert.cpp

```diff
@@ -175,10 +175,12 @@ namespace libtorrent {
 
 	std::string file_completed_alert::message() const
 	{
-		char msg[200 + TORRENT_MAX_PATH];
-		std::snprintf(msg, sizeof(msg), "%s: file %d finished downloading"
-			, torrent_alert::message().c_str(), static_cast<int>(index));
-		return msg;
+		std::string ret { torrent_alert::message() };
+		char msg[200];
+		std::snprintf(msg, sizeof(msg), ": file %d finished downloading"
+			, static_cast<int>(index));
+		ret.append(msg);
```

Comment:
Do you think there's some of this code that can be refactored, to perhaps have a net decrease of lines of code?